### PR TITLE
[posix] use enum for ot-ctl CLI parsing states

### PR DESCRIPTION
**Pull request by simonlingoogle**
One commit for each state:
- PromptState
- DoneState
- ErrorState

### CODE REVIEW

File: src/posix/client.cpp [hunk 1]
```diff
@@ -76,95 +76,132 @@ static void InputCallback(char *aLine)
...
-static bool FindPrompt(int &aState, char aChar)
+enum PromptState
+{
+    kPromptStateNewline,
+    kPromptStateChar1Received,
+    kPromptStatePrompted,
+    kPromptStateWaitNewline,
+};
+
+static bool FindPrompt(PromptState &aState, char aChar)
```
Comment (bukepo) [inline src/posix/client.cpp:87]: how about implement a generic finder?

```cpp
bool Find(int &aState, const char* aPattern, char aChar)
{
    // ...
    if (aPattern[aState] == aChar) // move the next state
    else if (aPattern[aState == '\r' || '\n') // state to zero
    else // state to invalid
    // ...
}
```

Comment (simonlingoogle) [reply src/posix/client.cpp:87]: what do you mean by `generic finder` ?

File: src/posix/client.cpp [hunk 2]
```diff
@@ -76,95 +76,132 @@ static void InputCallback(char *aLine)
...
 {
     switch (aChar)
     {
     case '>':
-        aState = aState == 0 ? 1 : -1;
+        aState = aState == kPromptStateNewline ? kPromptStateChar1Received : kPromptStateWaitNewline;
```
Comment (jwhui) [inline src/posix/client.cpp:92]: For state machines, I would prefer if we "pivot" this switch statement to `switch (aState)`. It makes it easier to ensure we handle each state appropriately and the conditions for state transmissions from each state.

Comment (simonlingoogle) [reply src/posix/client.cpp:92]: Agree.

**Comment by simonlingoogle** [comment]
Close for now to favor a generic implementation.

**Pull request closed without merging**